connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```
@@ -0,0 +1,304 @@
+import static org.apache.kafka.connect.runtime.distributed.DistributedConfig.STATUS_STORAGE_REPLICATION_FACTOR_CONFIG;
+import static org.apache.kafka.connect.runtime.distributed.DistributedConfig.STATUS_STORAGE_TOPIC_CONFIG;
+
+/**
+ * Start an embedded connect worker. Internally, this class will spin up a Kafka and Zk cluster, setup any tmp
+ * directories and clean up them on them.
+ */
+public class EmbeddedConnectCluster {
+
+    private static final Logger log = LoggerFactory.getLogger(EmbeddedConnectCluster.class);
+
+    private static final int DEFAULT_NUM_BROKERS = 1;
+    private static final int DEFAULT_NUM_WORKERS = 1;
+    private static final Properties DEFAULT_BROKER_CONFIG = new Properties();
+    private static final String REST_HOST_NAME = "localhost";
+
+    private final Connect[] connectCluster;
+    private final EmbeddedKafkaCluster kafkaCluster;
+    private final Map<String, String> workerProps;
+    private final String connectClusterName;
+
+    private EmbeddedConnectCluster(String name, Map<String, String> workerProps, int numWorkers, int numBrokers, Properties brokerProps) {
+        this.workerProps = workerProps;
+        this.connectClusterName = name;
+        this.kafkaCluster = new EmbeddedKafkaCluster(numBrokers, brokerProps);
+        this.connectCluster = new Connect[numWorkers];
+    }
+
+    /**
+     * Start the connect cluster and the embedded Kafka and Zookeeper cluster.
+     */
+    public void start() throws IOException {
+        kafkaCluster.before();
+        startConnect();
+    }
+
+    /**
+     * Stop the connect cluster and the embedded Kafka and Zookeeper cluster.
+     * Clean up any temp directories created locally.
+     */
+    public void stop() {
+        for (Connect worker : this.connectCluster) {
+            try {
+                worker.stop();
+            } catch (Exception e) {
+                log.error("Could not stop connect", e);
```

we shouldn't just be swallowing these, it would cause broken worker stop code to pass tests. we can either collect them and throw a wrapper or just let this propagate.